### PR TITLE
[HIPIFY] docs fixes

### PR DESCRIPTION
**Pull request by Rmalavally**
Incorporated Leo's feedback

### CODE REVIEW

File: docs/tables/CUBLAS_API_supported_by_HIP.md [hunk 1]
```diff
@@ -1034,4 +1034,4 @@
 |`cublasLtHeuristicsCacheSetCapacity`|11.8| | | | | | | | | |
 
 
-\*A - Added; D - Deprecated; C - Changed; R - Removed; E - Experimental
```
Comment (emankov) [inline docs/tables/CUBLAS_API_supported_by_HIP.md:1037]: What has changed here?
Btw, all CUDA2HIP documentation is entirely generated by hipify-clang and is not intended to be manually changed.

**Review by emankov — CHANGES_REQUESTED** [review]
LGTM, except CUDA2HIP md documentation, which is auto-generated and is not intended to be changed manually.

File: docs/tables/CUBLAS_API_supported_by_HIP.md (hunk 1 above)
Comment (Rmalavally) [reply docs/tables/CUBLAS_API_supported_by_HIP.md:1037]: Cosmetic changes @emankov 

- Added the NVIDIA prefix to CUDA on each page at the first instance
- Aligned ToC for Supported APIs page with the table

Comment (emankov) [reply docs/tables/CUBLAS_API_supported_by_HIP.md:1037]: I meant only that particular line, which looks to me unchanged:

\*A - Added; D - Deprecated; C - Changed; R - Removed; E - Experimental

And anyway, all `CU***API_supported_by_(HIP|ROC)(_and_ROC)?.md` are autogenerated, so please remove them from the PR.

**Review by lpaoletti — APPROVED** [review]

**Pull request merged**